### ISSUE TITLE
Better identification of authors

### ISSUE THREAD

**Issue by niemela** [Kattis/problem-package-format#78]
(See also #67)

Currently we identify a problem author by their name as a string. We discussed allowing for a more unique identifier such as email or orcid.

This could look something like this:
```
author:
   - name: Firstname Lastname
   - email: firstname.lastname@email.com
   - orcid: https://orcid.org/0000-0001-5727-2427
```
`email` and `orcid` would be optional, and just a string would be interpreted as the `name`.

Thoughts?

**Comment by jsannemo** [comment]
So you would write something like:
```yaml
authors:
  - name: blah
    email: blah
  - name: blah
    email: blah
```
etc?

Seems kind of verbose. I won't use it, but it's metadata so I care much less about spec bloat.

I would slightly more often use the email format though, i.e.
```
authors:
  - Johan Sannemo <jsannemo@jsannemo.se>
  - Simon Lindholm <....>
 ```
would be ni

**Comment by RagnarGrootKoerkamp** [comment]
I like the conciseness `first last <email>`

**Comment by niemela** [comment]
> I like the conciseness `first last <email>`

Yes, this was what I was thinking before someone suggested that we should support ORCID. I don't know if that was actually a joke? (But it does make some sense).

We could of course allow `First Last <email> ORCID` where email and ORCID are both optional. I'm guessing that ORCID would not be used by most, and then it would look exactly like if we didn't support it, and the parsing would not be much worse.

Would anybody use ORCID if it was supported?

**Comment by RagnarGrootKoerkamp** [comment]
I'm not exactly sure what the purpose of specifying email/orcid is anyway. As a more unique identifier than name only? Do you want to show them in kattis UI? Orcid is very research focussed, and maybe (probably) too unrelated to what we're doing (but sure, it would work as an ID, when provided)

**Comment by niemela** [comment]
> I'm not exactly sure what the purpose of specifying email/orcid is anyway. As a more unique identifier than name only?

Yes, exactly.

> Do you want to show them in kattis UI? 

The email/ORCID, no, but I would like to not merge problem authors that just happen to have the same name.

> Orcid is very research focussed, and maybe (probably) too unrelated to what we're doing (but sure, it would work as an ID, when provided)

Yeah, I have no idea how common it would be for problem authors to have an ORCID? Let's drop that t.b.h.

**Comment by RagnarGrootKoerkamp** [comment]
ok yes, let's drop ORCID and say something like 'authors with the exact same `first last` name are merged, unless they have conflicting email addresses', so that `ragnar groot koerkamp` and `ragnar groot koerkamp <me@gmail.com>` *do* get merged.

(Oh note: we should check if the string after the last whitespace is `<.*>`. If so, that's the email. Everything else, not matter how many words, is the name.)

**Comment by Tagl** [comment]
I like adding email to the metadata, although email addresses are not permanent. Also would like as a user to see it in UIs, such as Kattis, if present so that I know who to contact if the problem is broken in any way.

**Comment by niemela** [comment]
> Also would like as a user to see it in UIs, such as Kattis, if present so that I know who to contact if the problem is broken in any way.

We would certainly not do that by default. We could allow problem authors to turn that on though.

**Comment by Tagl** [comment]
> We would certainly not do that by default. We could allow problem authors to turn that on though.

I think that's completely fair. I can see why you wouldn't want it displayed.

**Comment by eldering** [comment]
I don't care too much, but I think that  the proposal of an author either consisting of a string, or a dictionary with `name`, `email`, `orcid`, `website`, or whatever other keys makes sense.
I'd add the option that a plain string can be in the `mailbox` format too: `John von Judge <john@judge.example.com>`.

**Comment by niemela** [comment]
> ok yes, let's drop ORCID and say something like 'authors with the exact same `first last` name are merged, unless they have conflicting email addresses', so that `ragnar groot koerkamp` and `ragnar groot koerkamp <me@gmail.com>` _do_ get merged.

I don't think we should specify exactly how a system uses this data, but that's probably approximately how I would do it.

**Comment by jsannemo** [comment]
There's a strong consensus for at least extending the name format to allowing Name <email>, and letting the previously comma separated list be a YAML sequence. I think updating the spec with that would be a good start @niemela ?

**Comment by niemela** [comment]
I consider this fixed by #106
